common/Windows/SecondaryWindowTemplate.xaml.cs

```diff
@@ -28,15 +28,15 @@ public object MainContent
     public SecondaryWindowTemplate(SecondaryWindow window)
     {
         _secondaryWindow = window;
-        this.InitializeComponent();
-    }
 
-    private void OnLoaded(object sender, RoutedEventArgs e)
-    {
         // A custom title bar is required for full window theme and Mica support.
         // https://docs.microsoft.com/windows/apps/develop/title-bar?tabs=winui3#full-customization
+        // Note: We need to add the title bar before the template loads, or else the title bar will flicker
+        // between the default Windows title bar and our own custom title bar.
         _secondaryWindow.ExtendsContentIntoTitleBar = true;
```

Comment:
Consider moving this after initialization (More specifically the second statement)

Reply:
updated thanks